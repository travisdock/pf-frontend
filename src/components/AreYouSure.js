import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.onOpen}
            >
            <div className="table-popup">
                <p>{this.props.message}</p>
                <button className="delete" onClick={this.props.handleDelete} >Delete</button>
                <button className="close" onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;