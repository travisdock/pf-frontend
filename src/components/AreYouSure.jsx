import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.askIfSure}
            >
            <div className="table-popup">
                <p>Are You Sure?</p>
                <button onClick={this.props.handleDelete} >Delete</button>
                <button onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;