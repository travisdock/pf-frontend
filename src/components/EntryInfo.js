import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button className="delete" onClick={this.props.askIfSure}>Delete Entry</button>
                <button className="update" onClick={this.props.openUpdateModal}>Update Entry</button>
                <button className="copy" onClick={() => this.props.copyEntry(this.props)}>Copy Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;