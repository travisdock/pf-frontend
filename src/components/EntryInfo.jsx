import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button onClick={this.props.askIfSure}>Delete Entry</button>
                <button onClick={this.props.openUpdateModal}>Update Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;