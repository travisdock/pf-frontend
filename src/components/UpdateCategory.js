import React from 'react';
import Popup from "reactjs-popup";

class UpdateCategory extends React.Component {
    render() {
        return (
            <Popup
                open={this.props.open === "update"}
                closeOnDocumentClick
                onOpen={this.props.onOpen}
                onClose={this.props.closeModal}
            >
                <div className="table-popup">
                <p>Update Category</p>
                <form onSubmit={this.props.handleUpdate}>
                    <input
                        name="name"
                        value={this.props.name}
                        onChange={this.props.handleChange}
                        >
                    </input>
                    <label>
                    <input
                        name="income"
                        type="checkbox"
                        checked={this.props.income}
                        onChange={this.props.toggleIncome}
                    />Income</label>
                    <label>
                    <input
                        name="untracked"
                        type="checkbox"
                        checked={this.props.untracked}
                        onChange={this.props.toggleUntracked}
                    />Untracked</label>
                    <input
                        name="year"
                        value={this.props.year}
                        onChange={this.props.handleChange}
                    />
                    <button type="submit" className="button">
                    Submit
                    </button>
                </form>

                </div>
            </Popup>
        )
    }
};

export default UpdateCategory;
