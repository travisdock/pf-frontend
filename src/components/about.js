import React from 'react';

const About = () => (
  <div className="about">
    <h1>Sum.</h1>
    <p>Some guy once said something like "What gets measured, gets managed". For people having trouble managing their personal finances, simply measuring their income and expenses is the first step. Often just the act of knowing where your money is going brings about the motivation to make changes. Sum. is a very simple app that gives the user a lot of freedom in simply measuring. Create any number of custom categories for income and expenses. Then watch where the money goes. Once you actually see how your expenses add up it will almost be harder not to make any changes.</p>
  </div>
)

export default About;
