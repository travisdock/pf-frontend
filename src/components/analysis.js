import React from 'react';
import { connect } from 'react-redux';

import Chart from './chart';

class Analysis extends React.Component {
  state = {
    userEntries: []
  }

  componentDidMount() {
    const url = `http://localhost:3001/api/v1/month_category/${this.props.current_user.user_id}`
    fetch(url)
      .then(resp => resp.json())
      .then(resp => this.setState({userData: resp}))
  }
  render(){

    return(
      <div>{this.state.userData ?
        <div><Chart /></div>
        :
        <div>Loading...</div>
      }</div>
    )
  }
}

const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Analysis);
