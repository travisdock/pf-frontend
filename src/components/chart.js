import React from 'react';
import {bb} from "billboard.js";
import { connect } from 'react-redux';

class Chart extends React.Component {

  state = {
    charts: {},
    currentChart: 0,
    stats: {},
    extraMonthInfo: {},
    load: false
  }

  componentDidMount() {
    const id = this.props.current_user.user_id

    fetch(`${process.env.REACT_APP_API}/api/v1/charts/${id}`)
      .then(resp => resp.json())
      .then(resp => {if (resp.error) {
        this.setState({error: resp.error})
      } else {
        const titles = Object.keys(resp.charts)
        const lastMonth = titles[titles.length - 1]
        this.setState({
          charts: resp.charts,
          stats: resp.stats,
          load: true,
          currentChart: resp.charts[lastMonth], currentTitle: lastMonth
        }, this.renderPieChart)
      }}
        )
  }

  selectChangePieChart = (e) => {
    const title = e.target.value
    this.setState( () => {
      return {
        currentChart: this.state.charts[title],
        currentTitle: title
      }
    }, this.renderPieChart)
  }

  renderPieChart() {
    const { currentChart, currentTitle } = this.state
    // const { currentTitle } = Object.keys(currentChart)[0]
    // debugger;
    bb.generate({
      data: {
        columns: currentChart,
        type: "pie",
        // onclick: function(d, element) { debugger }
      },
      title: {
        text: currentTitle,
        position: "top-center",
        padding: {
            top: 10,
           right: 10,
            bottom: 10,
            left: 10
        },
      },
      legend: {
        show: true,
        // contents: {
        //     bindto: "#legend",   // <ul id='legend'></ul>
        //     template: "<li style='background-color:#1f77b4'>data1</li>"
        //     or using function
        //     template: function(id, color, data) {
        //          // if you want omit some legend, return falsy value
        //          if (title !== "data1") {
        //               return "<li style='background-color:"+ color +">"+ title +"</li>";
        //          }
        //     }
        // },
        position: "bottom",  // bottom, right, inset
        equally: false,
        padding: 10,
        item: {
            // onclick: function(id) { debugger },
            // onover: function(id) { debugger },
            // onout: function(id) { debugger },
    
            // set tile's size
            tile: {
                width: 20,
                height: 15
            }
        }
      },
      bindto: "#chart"
    });
  }

  render() {
    return (
      <div className="content">
        {this.state.load ?
            <div className="chart-content">
              <select
                name="chart"
                value={this.state.currentTitle} //current chart month
                onChange={this.selectChangePieChart}
                >
                {Object.keys(this.state.charts).map(title => <option value={title} key={title} > {title} </option>) }
              </select>
              <div id="chart" />
            </div> :
            this.state.error ? <div>{this.state.error}</div> : <div>Loading...</div>
        }
      </div>
    )
  }
}
const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Chart)



// This is the profit loss chart. I got rid of it because it was ugly but I may want to remake it one day so here is the code.
// renderPLChart() {
//   const { currentChart } = this.state
//   bb.generate({
//     data: {
//       x: "x",
//       columns: currentChart,
//     type: "area-spline"
//     },
//     axis: {
//       x: {
//         type: "timeseries",
//         tick: {
//           format: "%b"
//         }
//       }
//     },
//     bindto: "#chart"
//   });
// }