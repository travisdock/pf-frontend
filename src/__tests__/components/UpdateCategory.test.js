import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { UpdateCategory } from '../../components/UpdateCategory';

const props = {
    open: 'update',
    onOpen: jest.fn(),
    closeModal: jest.fn(),
    name: 'categoryA',
    income: 'false',
    untracked: 'false',
    year: '1993',
    handleChange: jest.fn(),
    toggleIncome: jest.fn(),
    toggleUntracked: jest.fn(),
    handleUpdate: jest.fn()
};
const noProps = {
    open: 'false'
}

describe('snapshots', () => {
    it('renders an open <UpdateCategory /> snapshot', () => {
        const component = renderer.create(<UpdateCategory {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders a closed <UpdateCategory /> snapshot', () => {
        const component = renderer.create(<UpdateCategory {...noProps} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('update Category methods fire appropriately', () => {
    it('calls handleChange on category name input change', () => {
        const component = shallow(<UpdateCategory {...props} />);
        
        const input = component.find('input[name="name"]').first()
        input.simulate('change', {target: { value: 'new one' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });

    it('calls toggleIncome on checkbox check', () => {
        const component = shallow(<UpdateCategory {...props} />);
        
        const input = component.find('input[name="income"]').first()
        input.simulate('change', {target: { checked: true } });
        
        expect(props.toggleIncome).toHaveBeenCalled();
    });

    it('calls toggleUntracked on checkbox check', () => {
        const component = shallow(<UpdateCategory {...props} />);
        
        const input = component.find('input[name="untracked"]').first()
        input.simulate('change', {target: { checked: true } });
        
        expect(props.toggleUntracked).toHaveBeenCalled();
    });

    it('calls handleChange on year input change', () => {
        const component = shallow(<UpdateCategory {...props} />);
        
        const input = component.find('input[name="year"]').first()
        input.simulate('change', {target: { value: '1990' } });
        
        expect(props.handleChange).toHaveBeenCalled();
    });

    test('handleSubmit is fired on button click', () => {
        const component = shallow(<UpdateCategory {...props} />);

        const form = component.find('form')
        form.simulate('submit', { preventDefault () {} });

        expect(props.handleUpdate).toHaveBeenCalled();
    });
});