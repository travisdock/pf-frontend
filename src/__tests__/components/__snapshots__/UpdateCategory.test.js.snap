// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders a closed <UpdateCategory /> snapshot 1`] = `null`;

exports[`snapshots renders an open <UpdateCategory /> snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "left": "0px",
        "position": "absolute",
        "top": "0px",
      }
    }
  />,
  <div
    className="popup-overlay "
    onClick={[Function]}
    style={
      Object {
        "background": "rgba(0, 0, 0,0.5)",
        "bottom": "0",
        "display": "flex",
        "left": "0",
        "position": "fixed",
        "right": "0",
        "top": "0",
        "zIndex": "999",
      }
    }
  >
    <div
      className="popup-content "
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(255, 255, 255)",
          "border": "1px solid rgb(187, 187, 187)",
          "margin": "auto",
          "padding": "5px",
          "position": "relative",
          "width": "50%",
        }
      }
    >
      <div
        className="table-popup"
      >
        <p>
          Update Category
        </p>
        <form
          onSubmit={[MockFunction]}
        >
          <input
            name="name"
            onChange={[MockFunction]}
            value="categoryA"
          />
          <label>
            <input
              checked="false"
              name="income"
              onChange={[MockFunction]}
              type="checkbox"
            />
            Income
          </label>
          <label>
            <input
              checked="false"
              name="untracked"
              onChange={[MockFunction]}
              type="checkbox"
            />
            Untracked
          </label>
          <input
            name="year"
            onChange={[MockFunction]}
            value="1993"
          />
          <button
            className="button"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
